### PR TITLE
feat: add selective message inclusion when committing prompt sessions to versions

### PR DESCRIPTION
**Pull request by impoiler**
## Summary

Add selective message inclusion when committing prompt sessions as versions. Users can now choose which specific messages from a session to include in the committed version instead of always including all messages.

## Changes

- Added `message_indices` optional field to `CommitSessionRequest` to specify which messages to include (0-based indices)
- Enhanced backend handler to validate message indices and filter session messages accordingly
- Redesigned the commit version UI with a scrollable message selection interface
- Added message preview components with checkboxes for individual selection
- Implemented select all/deselect all functionality
- Added validation to prevent committing with no messages selected
- Fixed edge case where committing without changes now properly handles existing sessions

## Type of change

- [x] Feature

## Affected areas

- [x] Core (Go)
- [x] Transports (HTTP)
- [x] UI (Next.js)

## How to test

1. Create a prompt session with multiple messages
2. Click "Commit Version" to open the commit sheet
3. Verify you can select/deselect individual messages
4. Test the "Select all"/"Deselect all" toggle
5. Verify validation prevents committing with invalid indices
6. Confirm the committed version only contains selected messages

```sh
# Core/Transports
go version
go test ./...

# UI
cd ui
pnpm i || npm i
pnpm test || npm test
pnpm build || npm run build
```

## Screenshots/Recordings

The commit version sheet now displays a scrollable list of messages with checkboxes, allowing users to selectively include messages in the version.

## Breaking changes

- [ ] Yes
- [x] No

The `message_indices` field is optional, maintaining backward compatibility.

## Related issues

Enables more granular version control for prompt engineering workflows.

## Security considerations

Added validation to ensure message indices are within valid bounds to prevent out-of-range access.

## Checklist

- [x] I read `docs/contributing/README.md` and followed the guidelines
- [x] I added/updated tests where appropriate
- [x] I updated documentation where needed
- [x] I verified builds succeed (Go and UI)
- [x] I verified the CI pipeline passes locally if applicable

### CODE REVIEW

**Comment by impoiler** [comment]
* **#2524** <a href="https://app.graphite.com/github/pr/maximhq/bifrost/2524?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2132** <a href="https://app.graphite.com/github/pr/maximhq/bifrost/2132?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/maximhq/bifrost/2132?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#2375** <a href="https://app.graphite.com/github/pr/maximhq/bifrost/2375?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#2391** <a href="https://app.graphite.com/github/pr/maximhq/bifrost/2391?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `v1.5.0`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: ui/components/prompts/sheets/commitVersionSheet.tsx [hunk 1]
```diff
@@ -32,18 +78,50 @@ export function CommitVersionSheet({ open, onOpenChange, session, onCommitted }:
...
   async function onSubmit(data: CommitVersionFormData) {
     try {
+      const sortedIndices = Array.from(selectedIndices).sort((a, b) => a - b)
+      const commitData: { commit_message: string; message_indices?: number[] } = {
+        commit_message: data.commitMessage.trim(),
+      }
+      // Only send message_indices if not all messages are selected
+      if (!allSelected) {
+        commitData.message_indices = sortedIndices
+      }
```
[the remark this answers is not in the document]
Comment (impoiler) [reply ui/components/prompts/sheets/commitVersionSheet.tsx:124]: Added the validation

File: ui/components/prompts/sheets/commitVersionSheet.tsx [hunk 2]
```diff
@@ -32,18 +78,54 @@ export function CommitVersionSheet({ open, onOpenChange, session, onCommitted }:
...
+      if (next.has(index)) {
+        next.delete(index)
+      } else {
+        next.add(index)
+      }
+      return next
+    })
+  }, [])
+
+  const allSelected = selectedIndices.size === session.messages.length
```
[the remark this answers is not in the document]
Comment (impoiler) [reply ui/components/prompts/sheets/commitVersionSheet.tsx:101]: It’s very nit, and we don’t allow this sheet to be opened without messages.

**Comment by akshaydeo** [comment]
### Merge activity

* **Apr 6, 8:14 AM UTC**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.com/github/pr/maximhq/bifrost/2132).
* **Apr 6, 8:16 AM UTC**: [Graphite](https://app.graphite.com/github/pr/maximhq/bifrost/2132) couldn't merge this pull request because a downstack PR #2375 failed to merge.
* **Apr 6, 10:41 AM UTC**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.com/github/pr/maximhq/bifrost/2132).
* **Apr 6, 10:43 AM UTC**: @akshaydeo merged this pull request with [Graphite](https://app.graphite.com/github/pr/maximhq/bifrost/2132).
